sktime/transformers/rocket.py

```diff
@@ -0,0 +1,178 @@
+from numba import njit, prange
+import numpy as np
+import pandas as pd
+
+from .base import BaseTransformer
+
+__author__ = "Angus Dempster"
+__all__ = ["ROCKET"]
+
+class Rocket(BaseTransformer):
+    """ROCKET
+
+    RandOm Convolutional KErnel Transform
+
+    @article{dempster_etal_2019,
+      author  = {Dempster, Angus and Petitjean, Fran\c{c}ois and Webb, Geoffrey I},
+      title   = {ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels},
+      year    = {2019},
+      journal = {arXiv:1910.13051}
+    }
+
+    Parameters
+    ----------
+    input_length : int, length of input time series
+    num_kernels  : int, number of random convolutional kernels
+    num_channels : int, number of channels for multivariate time series (default 1)
+    """
+
+    def __init__(self,  input_length, num_kernels, num_channels = 1):
+        self.input_length = input_length
+        self.num_kernels = num_kernels
+        self.num_channels = num_channels
+        self.kernels = _generate_kernels(input_length, num_kernels, num_channels)
+
+    def fit(self, X, y = None):
+        return self
+
+    def fit_transform(self, X, y = None):
+        return self.fit(X).transform(X)
+
+    def transform(self, X):
+        """Transforms input time series using random convolutional kernels.
+
+        Parameters
+        ----------
+        X : pandas DataFrame, input time series (sktime format)
+
+        Returns
+        -------
+        pandas DataFrame, transformed features
+        """
+        return pd.DataFrame(_apply_kernels(self._to_numpy(X), self.kernels))
+
+    @staticmethod
+    def _to_numpy(X, a = None, b = None):
+        return np.stack(X.iloc[a:b].applymap(lambda cell : cell.to_numpy()).apply(lambda row : np.stack(row), axis = 1).to_numpy())
+
+@njit("Tuple((float32[:],int32[:],float32[:],int32[:],int32[:],int32[:],int32[:]))(int64,int64,int64)")
+def _generate_kernels(input_length, num_kernels, num_channels):
```

Comment:
Could we perhaps add this as a method to the transformer, so that we can use a random state similar to sklearn for reproducibility?

Reply:
`_generate_kernels` can be moved into the class easiliy (the other two functions are trickier for obscure reasons).  Either way, it should be possible to use a seed / random state.

Reply:
Ok, as long as we can have a random state in the constructor which is propagated to the functions, please use sklearn's `check_random_state` function :-)